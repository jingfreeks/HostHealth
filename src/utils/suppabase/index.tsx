export {default as supabase} from './suppabase';
