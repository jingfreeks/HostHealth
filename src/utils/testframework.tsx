import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
