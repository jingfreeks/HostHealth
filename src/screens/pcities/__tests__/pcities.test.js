import React from 'react';
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import {renderWithProviders} from '@/utils/testframeworknew';
import {waitFor} from '@testing-library/react-native';
import Pcities from '../pcities';

jest.useFakeTimers();
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      navigate: () => {},
    }),
    useIsFocused: () => true,
    useDispatch: () => ({dispatch: jest.fn()}),
  };
});
jest.mock('@supabase/supabase-js');
jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

describe('Popular Cities Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Pcities />);
    waitFor(() => {
      expect(all.toJSON()).toMatchSnapshot();
    });
  });
});
