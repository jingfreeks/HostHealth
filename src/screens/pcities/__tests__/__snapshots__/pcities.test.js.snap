// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular Cities Screen Should work as expected to get snapshot 1`] = `
<RCTScrollView
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignSelf": "stretch",
                "borderColor": "#E6E6E6",
                "borderRadius": 10,
                "borderWidth": 1,
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Image
              resizeMode="stretch"
              source={
                {
                  "uri": "https://img.freepik.com/free-photo/modern-tokyo-street-background_23-2149394948.jpg?w=996&t=st=1691132416~exp=1691133016~hmac=ee53d564abdbe32689b1a997ac989046fba50feb0f76dd84f912d8bea1a1121b",
                }
              }
              style={
                [
                  {
                    "borderBottomLeftRadius": 10,
                    "borderTopLeftRadius": 10,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "flexBasis": 0,
                  "flexGrow": 3,
                  "flexShrink": 1,
                  "marginBottom": 5,
                  "marginTop": 5,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "paddingBottom": 5,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 5,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#03606A",
                      "fontFamily": "Avenir",
                      "fontSize": 16,
                      "fontWeight": "900",
                    },
                  ]
                }
              >
                San Francisco
                 
                <Text
                  style={
                    [
                      {
                        "color": "#0e3f6c",
                        "fontFamily": "Avenir",
                        "fontSize": 12,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  San Francisco
                </Text>
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "paddingBottom": 5,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 5,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#0e3f6c",
                        "fontFamily": "Avenir",
                        "fontSize": 14,
                        "fontWeight": "900",
                      },
                    ]
                  }
                >
                  15
                   
                  <Text
                    style={
                      [
                        {
                          "color": "#0e3f6c",
                          "fontFamily": "Avenir",
                          "fontSize": 12,
                          "fontWeight": "500",
                        },
                      ]
                    }
                  >
                    Matches
                  </Text>
                </Text>
              </View>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "Avenir",
                      "fontSize": 16,
                      "fontWeight": "900",
                    },
                  ]
                }
              >
                $2,659
              </Text>
              <View
                style={
                  [
                    {
                      "alignSelf": "flex-end",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Avenir",
                        "fontSize": 12,
                        "fontWeight": "400",
                      },
                    ]
                  }
                >
                   /wk
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignSelf": "stretch",
                "borderColor": "#E6E6E6",
                "borderRadius": 10,
                "borderWidth": 1,
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Image
              resizeMode="stretch"
              source={
                {
                  "uri": "https://img.freepik.com/free-photo/los-angeles-downtown-buildings-night_649448-298.jpg?w=1060&t=st=1691142282~exp=1691142882~hmac=9a8088f4cc3274f95a936542763c9011320371fbc6b6534ee5f2d6e4108d8895",
                }
              }
              style={
                [
                  {
                    "borderBottomLeftRadius": 10,
                    "borderTopLeftRadius": 10,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "flexBasis": 0,
                  "flexGrow": 3,
                  "flexShrink": 1,
                  "marginBottom": 5,
                  "marginTop": 5,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "paddingBottom": 5,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 5,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#03606A",
                      "fontFamily": "Avenir",
                      "fontSize": 16,
                      "fontWeight": "900",
                    },
                  ]
                }
              >
                Boston
                 
                <Text
                  style={
                    [
                      {
                        "color": "#0e3f6c",
                        "fontFamily": "Avenir",
                        "fontSize": 12,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  Boston
                </Text>
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "paddingBottom": 5,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 5,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#0e3f6c",
                        "fontFamily": "Avenir",
                        "fontSize": 14,
                        "fontWeight": "900",
                      },
                    ]
                  }
                >
                  15
                   
                  <Text
                    style={
                      [
                        {
                          "color": "#0e3f6c",
                          "fontFamily": "Avenir",
                          "fontSize": 12,
                          "fontWeight": "500",
                        },
                      ]
                    }
                  >
                    Matches
                  </Text>
                </Text>
              </View>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "Avenir",
                      "fontSize": 16,
                      "fontWeight": "900",
                    },
                  ]
                }
              >
                $2,659
              </Text>
              <View
                style={
                  [
                    {
                      "alignSelf": "flex-end",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Avenir",
                        "fontSize": 12,
                        "fontWeight": "400",
                      },
                    ]
                  }
                >
                   /wk
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
